Common/src/Pollster/Agent.cs

```diff
@@ -116,27 +121,90 @@ public async Task FinalizeTaskCreation(CancellationToken cancellationToken)
 
     logger_.LogDebug("Finalize child task creation");
```

Comment:
I think the name of the method is wrong now, as this method implements much more than just finalizing task creation. Maybe it could be split into multiple methods. I think you could call directly the submitter steps instead of reimplementing them.

Reply:
I agree, the name does not show what is done. I will change it.

I already use what is available. I even already use Submitter itself.